toopher/toopher_iframe.py

```diff
@@ -0,0 +1,191 @@
+        api_uri = api_uri if api_uri else DEFAULT_BASE_URL
+        self.base_uri = api_uri.rstrip('/')
+
+    def get_user_management_url(self, username, reset_email, **kwargs):
+        if not 'ttl' in kwargs:
+            ttl = DEFAULT_IFRAME_TTL
+        else:
+            ttl = kwargs.pop('ttl')
+
+        params = {
+                'v':IFRAME_VERSION,
+                'username':username,
+                'reset_email':reset_email
+                }
+        params.update(kwargs)
+
+        return self._get_oauth_signed_url(self.base_uri + '/web/manage_user', params, ttl)
+
+    def get_authentication_url(self, username, reset_email='None', request_token='None', action_name='Log In', requester_metadata='None', **kwargs):
+        if not 'ttl' in kwargs:
+            ttl = DEFAULT_IFRAME_TTL
+        else:
+            ttl = kwargs.pop('ttl')
+
+        params = {
+            'v':IFRAME_VERSION,
+            'username':username,
+            'reset_email':reset_email,
+            'action_name':action_name,
+            'session_token':request_token,
+            'requester_metadata':requester_metadata
+        }
+        params.update(kwargs)
+
+        return self._get_oauth_signed_url(self.base_uri + '/web/authenticate', params, ttl)
+
+    def process_postback(self, urlencoded_form_data, request_token=None, **kwargs):
+        toopher_data = self._urldecode_iframe_data(urlencoded_form_data)
+
+        if 'error_code' in toopher_data:
+            error_message = toopher_data['error_message']
+            error = error_codes_to_errors[int(toopher_data['error_code'])]
+            raise error(error_message)
+        else:
+            validated_data = self._validate_data(toopher_data, request_token, kwargs)
+            toopher_api = ToopherApi(self.key, self.secret)
+            resource_type = validated_data['resource_type']
+            if resource_type == 'authentication_request':
+                return AuthenticationRequest(self._create_authentication_request_dict(validated_data), toopher_api)
+            elif resource_type == 'pairing':
+                return Pairing(self._create_pairing_dict(validated_data), toopher_api)
+            elif resource_type == 'requester_user':
+                return User(self._create_user_dict(validated_data), toopher_api)
+            else:
+                raise ToopherApiError('The postback resource type is not valid: {0}'.format(resource_type))
+
+    def is_authentication_granted(self, data, request_token=None, **kwargs):
+        try:
+            authentication_request = self.process_postback(data, request_token)
+            if isinstance(authentication_request, AuthenticationRequest):
+                return True if authentication_request.granted and not authentication_request.pending else False
+            else:
+                logging.warning('The postback did not return an AuthenticationRequest')
+                return False
+        except Exception as e:
```

Comment:
For the purposes of `is_authentication_granted`, should a `UserDisabledError` be treated differently?  If if the intention is for the user disabled flag to be used as an opt-out feature, then the user should be granted access, right?